src/ModelContextProtocol/Utils/Json/McpJsonUtilities.cs

```diff
@@ -68,8 +68,12 @@ internal static bool IsValidMcpToolSchema(JsonElement element)
                 {
                     return false;
                 }
-
-                return true; // No need to check other properties
```

Comment:
Now we need to update the `return false;` at the end of the loop to return true if the "type" property was found and know unknown properties were found.